fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -712,39 +809,136 @@ def tile_extent_from_rank_metadata(
         dims: dimension names
         rank_extent: the extent of one rank
         layout: the (y, x) number of ranks along each tile axis
+        edge_interior_ratio (optional): ratio between interior and boundary tile sizes.
+            Assumes full integer divisibility for both interior
+            and boundary tile sizes. Default is 1.0, meaning equal subtiles.
 
     Returns:
         tile_extent: the extent of one tile
     """
+    if edge_interior_ratio < 1.0:
+        raise NotImplementedError(
+            "Only equal sized subdomains are supported, was given "
+            f"an edge_interior_ratio of {edge_interior_ratio}"
+        )
     layout_factors = np.asarray(
         utils.list_by_dims(dims, layout, non_horizontal_value=1)
     )
     return extent_from_metadata(dims, rank_extent, layout_factors)
 
 
-def rank_extent_from_tile_metadata(
-    dims: Sequence[str], tile_extent: Sequence[int], layout: Tuple[int, int]
+def listArgsToTupleArgs(function):
```

Comment:
Please follow pothole_case for function names (including decorators).

Neat idea! I believe this will currently break if I call `subtile_extents_from_tile_metadata(dims=dims, tile_extent=tile_extent, layout=layout, edge_interior_ratio=ratio)`. Can you fix it to work with kwargs as well? You may want to write a unit test for it.

Reply:
kwargs support + accompanying unit test added.